### PR TITLE
upgrade to PostGIS 3.4

### PR DESCRIPTION
**Pull request by ImreSamu**
# Upgrade to PostGIS 3.4 

Changes from upstream:
* https://postgis.net/2023/08/PostGIS-3.4.0/
   * PostGIS version 3.4.0 has been released. This version is compatible with PostgreSQL versions 12-16.
* [pgsql-pkg-debian: postgis updated to version 3.4.0+dfsg-1.pgdg+1](https://www.postgresql.org/message-id/E1qWbgY-000eQB-DW%40atalia.postgresql.org) - 
    * The package postgis was updated on [apt.postgresql.org](http://apt.postgresql.org/).
    
The following updates have been made:

* For PostgreSQL versions 12-16, PostGIS has been upgraded from version 3.3 to 3.4, using the `git mv` command.
* The Dockerfiles have been updated using the `./update.sh` script.
* The `README.md` file has been updated manually.
* In the `Makefile`, the `LATEST_VERSION` has been manually updated to `15-3.4`.
   *  `VERSION=15-3.4 VARIANT=default make -n push`  looks OK
* The `.github/workflows/main.yml` file has been updated manually.

PostgreSQL 11 is not compatible with PostGIS 3.4, so it remains on PostGIS 3.3.

### CODE REVIEW

**Comment by ImreSamu** [comment]
Ready for review and merging.

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
 
-The `postgis/postgis` image provides tags for running Postgres with [PostGIS](http://postgis.net/) extensions installed. This image is based on the official [`postgres`](https://registry.hub.docker.com/_/postgres/) image and provides debian and alpine variants for PostGIS 3.3.x for each supported version of Postgres (11, 12, 13, 14 and 15).  Additionally, an image version is provided which is built from the latest two versions of Postgres (14, 15) with versions of PostGIS and its dependencies built from their respective master branches.
+The `postgis/postgis` image provides tags for running Postgres with [PostGIS](http://postgis.net/) extensions installed. This image is based on the official [`postgres`](https://registry.hub.docker.com/_/postgres/) image and provides debian and alpine variants for PostGIS 3.4.x for each supported version of Postgres (11, 12, 13, 14 and 15).  Additionally, an image version is provided which is built from the latest two versions of Postgres (14, 15) with versions of PostGIS and its dependencies built from their respective master branches.
```
Comment (phillipross) [inline README.md:5]: A little more clarity might be good in this paragraph, as it currently reads as though PostGIS 3.4.x is available for Postgres 11, when I believe 3.3.x is still used for V11.  I think we should spell out the special case here in addition to the correct details in the tables further down in the doc.

**Review by phillipross — CHANGES_REQUESTED** [review]
Looks good aside from suggested change needed for the 3.3.x for pgsql11 !

File: README.md (hunk 1 above)
Comment (ImreSamu) [reply README.md]: Good suggestion,  
I have created a new proposal.

**Comment by ImreSamu** [comment]
README improved.
and "All checks have passed".

Ready for review and merging.

**Review by phillipross — APPROVED** [review]
lgtm, thanks!

**Pull request merged**